cmd/terramate/cli/cli.go

```diff
@@ -988,34 +992,56 @@ func (c *cli) triggerStack(stack string) {
 	if reason == "" {
 		reason = "Created using Terramate CLI without setting specific reason."
 	}
-
-	if !path.IsAbs(stack) {
-		stack = filepath.Join(c.wd(), filepath.FromSlash(stack))
+	if !path.IsAbs(basePath) {
+		basePath = filepath.Join(c.wd(), filepath.FromSlash(basePath))
 	} else {
-		stack = filepath.Join(c.rootdir(), filepath.FromSlash(stack))
+		basePath = filepath.Join(c.rootdir(), filepath.FromSlash(basePath))
 	}
-
-	stack = filepath.Clean(stack)
-	_, err := os.Lstat(stack)
+	basePath = filepath.Clean(basePath)
+	_, err := os.Lstat(basePath)
 	if errors.Is(err, os.ErrNotExist) {
-		fatalWithDetails(err, "stack not found")
+		fatalWithDetails(err, "path not found")
 	}
-	tmp, err := filepath.EvalSymlinks(stack)
+	tmp, err := filepath.EvalSymlinks(basePath)
 	if err != nil {
 		fatalWithDetails(err, "failed to evaluate stack path symlinks")
 	}
-	if tmp != stack {
-		fatal(stdfmt.Sprintf("symlinks are disallowed in the stack path: %s links to %s", stack, tmp))
+	if tmp != basePath {
+		fatal(stdfmt.Sprintf("symlinks are disallowed in the path: %s links to %s", basePath, tmp))
 	}
-	if !strings.HasPrefix(stack, c.rootdir()) {
-		fatalf("stack %s is outside project", stack)
+	if !strings.HasPrefix(basePath, c.rootdir()) {
+		fatalf("stack %s is outside project", basePath)
 	}
-	stackPath := prj.PrjAbsPath(c.rootdir(), stack)
-	if err := trigger.Create(c.cfg(), stackPath, kind, reason); err != nil {
-		fatalWithDetails(err, "unable to create trigger")
+	prjBasePath := prj.PrjAbsPath(c.rootdir(), basePath)
+	if c.parsedArgs.Experimental.Trigger.Status != "" && c.parsedArgs.Experimental.Trigger.Recursive {
+		fatal("cloud filters like --status is incompatible with --recursive flag")
+	}
+	var stacks config.List[*config.SortableStack]
+	if !c.parsedArgs.Experimental.Trigger.Recursive {
+		st, found, err := config.TryLoadStack(c.cfg(), prjBasePath)
+		if err != nil {
+			fatalWithDetails(err, "loading stack in current directory")
+		}
+		if !found {
+			fatal("path is not a stack and --recursive is not provided")
+		}
+		stacks = append(stacks, st.Sortable())
+	} else {
+		var err error
+		stacksReport, err := c.listStacks(false, "", cloudstack.NoFilter)
```

Comment:
```suggestion
		stacksReport, err := c.listStacks(false, cloudstack.AnyTarget, cloudstack.NoFilter)
```
You can use `cloudstack.AnyTarget` here which makes it a bit more expressive what "" means. (I may have forgotten to replace this everywhere)